code/game/objects/machinery/squad_supply/supply_beacon.dm

```diff
@@ -114,6 +114,11 @@
 	SIGNAL_HANDLER
 	beacon_datum = null
 
+/obj/item/beacon/supply_beacon/onTransitZ(old_z,new_z)
+	. = ..()
+	//Assumes doMove sets loc before onTransitZ
```

Comment:
It does 
https://github.com/SiegeB0t/TerraGov-Marine-Corps/blob/d92507c2d393a6dcb0752ecad22754adce2a1867/code/game/atoms_movable.dm#L427